### ISSUE TITLE
`(p/all [nil])` fails to run in ClojureScript

### ISSUE THREAD

**Issue by Zatura24** [funcool/promesa#140]
I was toying with `p/plet` to make parallel calls to an external API in our ClojureScript project. However `p/plet` fails to run when it contains a `nil` binding. The underlying problem seems to be with `p/all`.

Example:

```clojure
@(p/plet [a (p/promise 1)
          b (when false (p/promise 2))]
   (prn a b))
;; nothing is printed

;; p/plet converts bindings to a p/all statement
@(p/all [(p/promise 1) nil])
;; nothing is returned

;; wrapped in a catch statement shows the error
@(p/catch 
   (p/all [(p/promise 1) nil]) 
   (fn [err] err))
;; #object[TypeError TypeError: Cannot read properties of null (reading 'then')]
```

This error comes from the `coerse` function inside the Javascript impl/promise.js file. Diving into the code, the `all` function for the ClojureScript variant seems to be missing the coersion to promises like in the Clojure variant.
https://github.com/funcool/promesa/blob/69dcb52a140a4237054ab6340048d5278bce680b/src/promesa/impl.cljc#L63-L70 

Therefore my proposed solution would be to give the `:cljs` variant the same treatment as the `:clj` variant like so
```clojure
(defn all
  [promises]
  #?(:cljs (let [promises (map pt/-promise promises)]
             (-> (impl/all (into-array promises))
                 (pt/-fmap vec)))
     :clj (let [promises (map pt/-promise promises)]
            (-> (CompletableFuture/allOf (into-array CompletableFuture promises))
              (pt/-fmap (fn [_] (mapv pt/-extract promises)))))))
```
Now `nil` values will be coersed to `(p/resolved nil)` thanks to the following code already being available:
https://github.com/funcool/promesa/blob/69dcb52a140a4237054ab6340048d5278bce680b/src/promesa/impl.cljc#L353-L361

Let me know if this is not a proper solution or if I'm missing anything

**Comment by aiba** [comment]
@Zatura24 your proposed solution seems good.  Do you happen to have this in a forked repo?  It would be great to submit this as a PR and also have a repo we could depend on in the meantime to work around this issue.  Thank you for your investigation!

**Comment by Zatura24** [comment]
@aiba I don't have my own fork for this. I might create one this weekend. Since it's a known issue for us, we've simply resolved it by explicitly returning a `nil` promise. So in my example turning `(when false (p/promise 2))` into `(if false (p/promise 2) (p/resolved nil))`.

**Comment by niwinz** [comment]
I will look on it this week

**Comment by aiba** [comment]
@Zatura24 thank you.  I suppose we could also work around this by call `pt/-promise`, i.e.:

```clojure
@(p/plet [a (p/promise 1)
          b (pt/-promise (when false (p/promise 2)))]
   (prn a b))
```

@niwinz thank you for taking a look!

**Comment by aiba** [comment]
Hi @niwinz just wanted to see if you were able to look into this.  This issue is preventing us from upgrading since we were not able to track down 100% of the p/lets and p/alls in our codebase and wrap them in the workaround.  Thank you!

**Comment by aiba** [comment]
I believe this bug only pertains to the `:cljs` clause of `p/all`, as in, it only manifests in clojurescript.

**Comment by niwinz** [comment]
OK, i was confused by the examples, I looking on it right now.

**Comment by niwinz** [comment]
Released `11.0.678` it should fix this issue.

**Comment by Zatura24** [comment]
Thanks for looking in to it!
